pkg/util/mathutil/math.go

```diff
@@ -54,21 +51,21 @@ func StrLenOfUint64Fast(x uint64) int {
 	}
 }
 
-// StrLenOfInt64Fast efficiently calculate the string character lengths of an int64 as input
+// StrLenOfInt64Fast efficiently calculates the length of the string representation of an int64.
 func StrLenOfInt64Fast(x int64) int {
 	size := 0
 	if x < 0 {
-		size = 1 // add "-" sign on the length count
+		size = 1 // for the negative sign
 	}
 	return size + StrLenOfUint64Fast(uint64(Abs(x)))
 }
 
-// IsFinite reports whether f is neither NaN nor an infinity.
+// IsFinite checks whether a float64 is neither NaN nor infinity.
 func IsFinite(f float64) bool {
-	return !math.IsNaN(f - f)
+	return !math.IsNaN(f) && !math.IsInf(f, 0)
 }
 
-// Max returns the largest one from its arguments.
+// Max returns the largest element from its arguments.
 func Max[T constraints.Ordered](x T, xs ...T) T {
```

Comment:
The original issue wants to remove this function. Caller can use the built-in `max` function instead